quickstart-docker/compose.yaml

```diff
@@ -0,0 +1,121 @@
+version: '2.1'
+services:
+
+  zookeeper:
+    image: confluentinc/cp-zookeeper:6.1.1
+    ports:
+      - "2187:2187"
+    environment:
+      ZOOKEEPER_CLIENT_PORT: 2187
+
+  # reachable on 9094 from the host and on 29094 from inside docker compose
+  kafka:
+    image: confluentinc/cp-kafka:6.1.1
+    depends_on:
+      - zookeeper
+    ports:
+      - '9094:9094'
+    expose:
+      - '29094'
+    environment:
+      KAFKA_ZOOKEEPER_CONNECT: 'zookeeper:2187'
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+      KAFKA_INTER_BROKER_LISTENER_NAME: PLAINTEXT
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka:29094,PLAINTEXT_HOST://localhost:9094
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: '1'
+      KAFKA_MIN_INSYNC_REPLICAS: '1'
+
+  init-kafka:
+    image: confluentinc/cp-kafka:6.1.1
+    depends_on:
+      - kafka
+    entrypoint: [ '/bin/sh', '-c' ]
+    command: |
+      "
+      # blocks until kafka is reachable
+      kafka-topics --bootstrap-server kafka:29094 --list
```

Comment:
Perhaps you can use the `healthcheck` directive instead ? Its meant exactly for these purposes.

You can see [here](https://www.baeldung.com/linux/docker-compose-container-interdependence#1-the-healthcheck-directive) for its explanation. For a sample yaml, check [here](https://github.com/peter-evans/docker-compose-healthcheck/blob/master/docker-compose.yml#L11)

Reply:
Also, is it necessary to create a different container to create the topics ? Can't we fire these from the kafka container itself ?

Reply:
for this we need different container because the kafka container uses yumm package manager and also the sudo gives error `sudo: command not found` so, instead i've added a check in the init-kafka container itself to check if the kafka is running or not